kmath-geometry/src/commonMain/kotlin/scientifik/kmath/geometry/RealEuclideanVectorSpace.kt

```diff
@@ -0,0 +1,75 @@
+package scientifik.kmath.geometry
+
+import scientifik.kmath.dimensions.*
+import scientifik.kmath.operations.ExtendedField
+import scientifik.kmath.operations.RealField
+import kotlin.math.sqrt
+
+data class RealVector<D : Dimension>(private val array: DoubleArray) : Vector<Double, D> {
+    override val size: Int = array.size
+
+    override fun get(index: Int): Double = when {
+        index < 0 || index >= size -> error("Accessing outside of point bounds")
+        else -> array[index]
+    }
+
+    override fun iterator(): Iterator<Double> = array.iterator()
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other == null || this::class != other::class) return false
+
+        other as RealVector<*>
+
+        return array.contentEquals(other.array)
+    }
+
+    override fun hashCode(): Int = array.contentHashCode()
+
+    override fun toString(): String = array.contentToString()
```

Comment:
Must be replaced with `"RealVector(${array.contentToString()})"`

Reply:
done